### ISSUE TITLE
Cloud native database compatibility

### ISSUE THREAD

**Issue by corporatepiyush** [JetBrains/Exposed#181]
Is it compatible with Google Cloud Spanner and CockroachDB ?

**Comment by Tapac** [comment]
No. Noone asked for that, so there are no such dialects in Exposed. I mark this issue as a feature request, but can't give you any estimates.

**Comment by dmcg** [comment]
We are running against Cockroach using PostgreSQL drivers and things are fine. You may need to write code to automagically retry, as Cockroach bails out of many more transactions than Postgres

**Comment by Jerbell** [comment]
I'm trying to run our Postgres app against Cockroach.  Not much success at the moment.  Currently having problems with altering columns, altering columns with an index and collation.

**Comment by matheusfrancisco** [comment]
Is there progress in adding google cloud spanner support? I would like to help if it is possible.

**Comment by Tapac** [comment]
@matheusfrancisco feel free to publish a PR with the GC Spanner support. Please use https://github.com/GoogleCloudPlatform/cloud-spanner-emulator for tests.

**Comment by lfgcampos** [comment]
> Is there any progress in adding google cloud spanner support? I would like to help if it is possible.

Hi @matheusfrancisco, any news on this? I am also interested in exposed and Google cloud spanner support and could potentially help if there is an on going effort, if needed!

**Comment by marcosb** [comment]
+1 - would be great to see `VendorDialect` for `com.google.cloud.spanner.jdbc.JdbcDriver`

**Comment by marcosb** [comment]
FWIW, `Database.registerJdbcDriver("jdbc:cloudspanner", "com.google.cloud.spanner.jdbc.JdbcDriver", PostgreSQLDialect.dialectName)` probably gets something which is mostly functional for Cloud Spanner in the interim

**Comment by alex-shinn** [comment]
Attempting that workaround gives:

```
com.google.cloud.spanner.jdbc.JdbcSqlExceptionFactory$JdbcSqlFeatureNotSupportedException: Only isolation level TRANSACTION_SERIALIZABLE is supported
```

**Comment by obabichevjb** [comment]
I tried to add Spanner to Exposed, but it looks like it's not so straightforward and requires some extra work and careful implementation of the dialect inside Exposed. I would suggest creating separate issues for particular databases if there is still demand.
